web/src/beta/features/ProjectSettings/innerPages/PluginSettings/hooks.ts

```diff
@@ -0,0 +1,186 @@
+import { useApolloClient } from "@apollo/client";
+import { useCallback, useMemo } from "react";
+
+// TODO: migrate to services API
+import {
+  useGetInstalledPluginsQuery,
+  useInstallPluginMutation,
+  useUninstallPluginMutation,
+  useUploadPluginMutation,
+  useUpgradePluginMutation,
+} from "@reearth/classic/gql/graphql-client-api";
+import { useT, useLang } from "@reearth/services/i18n";
+import { useNotification } from "@reearth/services/state";
+
+export type Plugin = {
+  fullId: string;
+  id: string;
+  version: string;
+  title?: string;
+  author?: string;
+};
+
+export default ({ projectId, sceneId }: { projectId: string; sceneId?: string }) => {
+  const [installPluginMutation] = useInstallPluginMutation();
+  const [uploadPluginMutation] = useUploadPluginMutation();
+  const [uninstallPluginMutation] = useUninstallPluginMutation();
+  const [upgradePluginMutation] = useUpgradePluginMutation();
+
+  const t = useT();
+  const lang = useLang();
+  const client = useApolloClient();
+  const [, setNotification] = useNotification();
+
+  const { data: rawSceneData, loading } = useGetInstalledPluginsQuery({
+    variables: { projectId: projectId ?? "", lang },
+    skip: !projectId,
+  });
+
+  const marketplacePlugins = useMemo(
+    () =>
+      rawSceneData?.scene?.plugins
+        .filter(p => p.plugin && p.plugin?.id !== "reearth" && p.plugin.id.split("~", 3).length < 3)
+        .map((p): Plugin | undefined => {
+          if (!p.plugin) return;
+          const fullId = p.plugin.id;
+          const [id, version] = fullId.split("~", 2);
+          return {
+            fullId,
+            id,
+            version,
+            title: p.plugin.name,
+            author: p.plugin.author,
+          };
+        })
+        .filter((p): p is Plugin => !!p) ?? [],
+    [rawSceneData],
+  );
+
+  const personalPlugins = useMemo(() => {
+    return (
+      rawSceneData?.scene?.plugins
+        .filter(p => p.plugin && p.plugin.id !== "reearth" && p.plugin.id.split("~", 3).length == 3)
+        .map(p => ({
+          title: p.plugin?.translatedName ?? "",
+          bodyMarkdown: p.plugin?.translatedDescription ?? "",
+          author: p.plugin?.author ?? "",
+          isInstalled: true,
+          pluginId: p.plugin?.id ?? "",
+        })) ?? []
+    );
+  }, [rawSceneData]);
```

Comment:
Please move this (and other appropriate variables/callbacks) in to services/api/plugins

Reply:
Updated.